src/flask_pyoidc/flask_pyoidc.py

```diff
@@ -491,11 +463,76 @@ def wrapper(*args, **kwargs):
                     logger.info('user has valid access token')
                     # Store token introspection info within the application
                     # context.
-                    g.token_introspection_info = token_introspection_result.to_dict()
+                    g._token_introspection_info = token_introspection_result.to_dict()
                     return view_func(*args, **kwargs)
                 # Forbid access if the access token is invalid.
-                flask.abort(403)
+                flask.abort(401)
```

Comment:
This should still be a 403 - a token was present in the request but it was not valid.